kernel/sys_clock.c

```diff
@@ -494,13 +494,13 @@ s32_t _get_next_timeout_expiry(void)
  * Must be called with interrupts locked.
  */
 
-void _add_timeout(struct k_thread *thread, struct _timeout *timeout,
-		  s32_t timeout_in_ticks)
+void _add_timeout(struct _timeout *timeout,
+		  _timeout_func_t fn, s32_t timeout_in_ticks)
 {
 	__ASSERT(timeout_in_ticks >= 0, "");
+	__ASSERT(fn == timeout->func, "");
```

Comment:
We do!  :)  I moved the callback to be an argument to the add() routine instead of init().  This assertion was added to be sure I didn't break something.